server/quota/namespace.go

```diff
@@ -0,0 +1,225 @@
+package quota
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	"github.com/tigrisdata/tigris/server/config"
+	"github.com/tigrisdata/tigris/server/metadata"
+	"go.uber.org/atomic"
+	"golang.org/x/time/rate"
+)
+
+const (
+	rateHysteresis       = 10         // ±10 rps per instance wouldn't cause rate regulation
+	throughputHysteresis = 100 * 1024 // ±100Kb per instance wouldn't cause throughput regulation
+
+	rateIncrement       = 5     // when rate adjustment is needed increment current rate by ± this value
+	throughputIncrement = 10000 // when throughput adjustment is needed increment by ± this value
```

Comment:
don't we need to adjust it by (+-)10% of the rate?

Reply:
This is good catch. Fixed.